File: log_parser.py
import os, datetime

def parse_between_time(_start, _end, filename="application-event-log.txt") -> int:
    '''
    Parses a windows event log given a start and end time counting how many
    entries were made that occurred between the start and end times. 

    returns: int: count # Of entries made
    '''
    # opening log file to parse and removing header line
    events = open(filename, "r")
    header = events.readline()

    # converting start and end into time objects
    start = datetime.datetime.strptime(_start, "%H:%M:%S").time()
    end = datetime.datetime.strptime(_end, "%H:%M:%S").time()

    # Comparing time of each entry to start and end, increment count if time inbetween
    count = 0
    for event in events:
        temp = event.split("\t")
        event_time = datetime.datetime.strptime(temp[2], "%I:%M:%S %p").time()
        
        if event_time <= end and event_time >= start:
            count +=1

    return count


test_start = "01:34:00"
test_end = "01:39:59"

test_count = parse_between_time(test_start, test_end)

print(f"Final count of entries between {test_start} and {test_end} was {test_count}")

File: log_parser.pl
use strict;
use warnings;
use Time::Piece;

sub parse_between_times {
    my ($s, $e, $file) = @_;
    # grabs total count of entries made between two given times for a given windows log file
    # inputs: 
    # s --> Start time: "01:34:00" "%H:%M:%S"
    # e --> End time: "01:39:59" "%H:%M:%S"
    # file --> filename: "application-event-log.txt"
    #
    # returns: 
    # count --> total count of entries between start and end time

    my $count = 0;
    
    # convert the times wanted based off military time. 
    my $start = Time::Piece->strptime($s, "%H:%M:%S");
    my $end = Time::Piece->strptime($e, "%H:%M:%S");
    
    # open file and get rid of headers
    open(my $file_handle, '<', $file) or die "Could not open '$file': $!";
    my $templ = <$file_handle>;

    # count number of entries between start and end times
    while (my $line = <$file_handle>) {
        my @columns = split /\t/, $line;
        my $time_object = Time::Piece->strptime($columns[2], "%I:%M:%S %p");
        if ($time_object >= $start && $time_object <= $end){
            $count++;
        }
    }

    return $count;
}

# Inputs
my $start_ = "01:34:00";
my $end_ = "01:39:59";
my $file_path_ = "application-event-log.txt";

my $count_entries = parse_between_times($start_, $end_, $file_path_);

# Output
print "Total count of log entries between $start_ and $end_ was --> $count_entries";
